src/hotspot/share/c1/c1_Instruction.hpp

```diff
@@ -2424,14 +2424,12 @@ class BlockPair: public CompilationResourceObj {
  private:
   BlockBegin* _from;
   BlockBegin* _to;
+  int _index;
```

Comment:
We don't really need `_to` anymore, because sux_at(_index) == _to if `insert_block_between` hasn't already processed this edge.  We can detect if sux_at(_index)  has been replaced in a couple different ways:
1. set a block flag on the newly inserted block
2. check the block_id on the newly inserted block, assume block ids increase monotonically.
But maybe the added complexity isn't worth it?  What's the performance improvement with this change vs the footprint increase for this new field?

Reply:
Inspired by your comment.
Only the method `BlockBegin::insert_block_between` sets `critical_edge_split_flag`, so we can check if that flag is set or not.
Then there is no footprint problem.